server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java

```diff
@@ -338,6 +361,207 @@ public void testWriteIncrementalMetadataSuccess() throws IOException {
+        final ClusterMetadataManifest previousManifest = ClusterMetadataManifest.builder()
+            .codecVersion(ClusterMetadataManifest.CODEC_V0)
+            .globalMetadataFileName(null)
+            .indices(Collections.emptyList())
+            .build();
+
+        remoteClusterStateService.start();
+        final ClusterMetadataManifest manifestAfterUpdate = remoteClusterStateService.writeIncrementalMetadata(
+            previousClusterState,
+            newClusterState,
+            previousManifest
+        );
+
+        // global metadata is updated
+        assertThat(manifestAfterUpdate.getGlobalMetadataFileName(), notNullValue());
+        // Manifest file with codec version with 1 is updated.
+        assertThat(manifestAfterUpdate.getCodecVersion(), is(ClusterMetadataManifest.CODEC_V1));
+    }
+
+    public void testWriteIncrementalGlobalMetadataSuccess() throws IOException {
+        final ClusterState clusterState = generateClusterStateWithGlobalMetadata().nodes(nodesWithLocalNodeClusterManager()).build();
+        mockBlobStoreObjects();
+        final CoordinationMetadata coordinationMetadata = CoordinationMetadata.builder().term(1L).build();
+        final ClusterState previousClusterState = ClusterState.builder(ClusterName.DEFAULT)
+            .metadata(Metadata.builder().coordinationMetadata(coordinationMetadata))
+            .build();
+
+        final ClusterMetadataManifest previousManifest = ClusterMetadataManifest.builder()
+            .codecVersion(2)
+            .globalMetadataFileName("global-metadata-file")
+            .indices(Collections.emptyList())
+            .build();
+
+        remoteClusterStateService.start();
+        final ClusterMetadataManifest manifest = remoteClusterStateService.writeIncrementalMetadata(
+            previousClusterState,
+            clusterState,
+            previousManifest
+        );
+
+        final ClusterMetadataManifest expectedManifest = ClusterMetadataManifest.builder()
+            .indices(Collections.emptyList())
+            .globalMetadataFileName("mock-filename")
+            .clusterTerm(1L)
+            .stateVersion(1L)
+            .stateUUID("state-uuid")
+            .clusterUUID("cluster-uuid")
+            .previousClusterUUID("prev-cluster-uuid")
+            .build();
+
+        assertThat(manifest.getGlobalMetadataFileName(), notNullValue());
+        assertThat(manifest.getClusterTerm(), is(expectedManifest.getClusterTerm()));
+        assertThat(manifest.getStateVersion(), is(expectedManifest.getStateVersion()));
+        assertThat(manifest.getClusterUUID(), is(expectedManifest.getClusterUUID()));
+        assertThat(manifest.getStateUUID(), is(expectedManifest.getStateUUID()));
+    }
+
+    /*
+     * Here we will verify global metadata is not uploaded again if change is only in index metadata
+     */
+    public void testGlobalMetadataNotUpdatingIndexMetadata() throws IOException {
```

Comment:
can we add assertion to existing indexMetadata test that globalMetadata is unchanged and not uploaded.

Reply:
if you are keeping this test then rename to testIndexMetadataOnlyUpdated or testGlobalMetadataOnlyUpdated for below test.

Reply:
Renamed it to `testIndexMetadataOnlyUpdated` and `testGlobalMetadataOnlyUpdated`